### PR TITLE
Ds mm

### PR DESCRIPTION
**Pull request by joechainz**

### CODE REVIEW

**Review by waynew — CHANGES_REQUESTED** [review]
One question - otherwise looks good!

File: src/saltext/vmware/states/datastore.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+__proxyenabled__ = ["vmware_datastore"]
+
+
+def __virtual__():
+    if not HAS_PYVMOMI:
+        return False, "Unable to import pyVmomi module."
+    return __virtualname__
+
+
+def maintenance_mode(name, maintenance_mode, datacenter_name=None, service_instance=None):
```
Comment (waynew) [inline src/saltext/vmware/states/datastore.py:29]: **question** would `set_maintenance_mode` be a better name? Also thinking it the argument could then be either renamed to `mode` or `activate_maintenance_mode` :thinking:

Comment (waynew) [reply src/saltext/vmware/states/datastore.py:29]: Actually, after a cursory glance at https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.resmgmt.doc/GUID-D229556E-C991-41BD-A378-363CC9D2B2CC.html

I think it makes sense to do:

```suggestion
def maintenance_mode(name, enter_maintenance_mode, datacenter_name=None, service_instance=None):
```

(and update the docstring to match)

File: src/saltext/vmware/states/datastore.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    service_instance
+        (optional) The Service Instance from which to obtain managed object references.
+    """
+    if service_instance is None:
+        service_instance = connect.get_service_instance(opts=__opts__, pillar=__pillar__)
+    ret = {"name": name, "changes": {}, "result": True, "comment": ""}
+    dc_ref = None
+    if datacenter_name:
+        dc_ref = utils_common.get_datacenter(service_instance, datacenter_name)
+    ds = utils_common.get_datastore(name, dc_ref, service_instance)
```
Comment (waynew) [inline src/saltext/vmware/states/datastore.py:51]: **question** does this correctly work when `dc_ref` is None?

File: src/saltext/vmware/states/datastore.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    status = ds.summary.maintenanceMode
+    if maintenance_mode:
+        if status == "inMaintenance":
+            ret["comment"] = "Already in maintenance mode."
+            return ret
+
+        if __opts__["test"]:
+            ret["changes"] = {"new": f"datastore {name} will enter maintenance mode."}
+            ret["comment"] = "These options are set to change."
```
Comment (waynew) [inline src/saltext/vmware/states/datastore.py:60]: **suggestion** missing `ret["result"] = None` here for test mode.

File: src/saltext/vmware/states/datastore.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        ret["result"] = False
+        return ret
+    else:
+        if status == "normal":
+            ret["comment"] = "Already exited maintenance mode."
+            return ret
+
+        if __opts__["test"]:
+            ret["changes"] = {"new": f"datastore {name} will exit maintenance mode."}
+            ret["comment"] = "These options are set to change."
```
Comment (waynew) [inline src/saltext/vmware/states/datastore.py:78]: **suggestion** missing `ret["result"] = None` here for test mode

**Review by waynew — CHANGES_REQUESTED** [review]
Looks good - just missing a couple of bits in the test states

File: src/saltext/vmware/states/datastore.py (hunk 2 above)
Comment (joechainz) [reply src/saltext/vmware/states/datastore.py:51]: yes

**Review by waynew — APPROVED** [review]
:rocket: nice!

**Pull request merged**